### PR TITLE
8 0 l10n be coda sequence

### PR DESCRIPTION
**Pull request by eezeefho**
Description of the issue/feature this PR addresses:
This module adds a checkbox on accounting journals of type Bank and Checks. It adds the ability to use the journal sequence for CODA import.
The External Bank Statement number (referenced in the CODA file) is put in a separate field.
A constraint has been added on the combination of Journal and External Bank Statement Number in order to avoid that a user would be able to import the same CODA file twice.

Current behavior before PR:
You can import the same CODA file

Desired behavior after PR is merged:
This module adds a checkbox on accounting journals of type Bank and Checks. It adds the ability to use the journal sequence for CODA import.
The External Bank Statement number (referenced in the CODA file) is put in a separate field.
A constraint has been added on the combination of Journal and External Bank Statement Number in order to avoid that a user would be able to import the same CODA file twice.
## 

I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by qdp-odoo** [comment]
Hi Francois, and thanks for the PR.

I've read through the diff and here is a list of things I'd like to comment:
* no need of compatibility check at installation: that's a bit overkill, don't bother
* no need of boolean 'Force Sequence' on journal: by installing the module, you already said you wanted that behavior. Keep it simple
* new field on bank.statement.line is a related (and should be written accordingly), but actually I don't see why you added it. It's useless, according to me, and could be simply removed
* I don't see any use of the demo data somewhere. I suppose you could remove them too and rely on the demo file that is already in l10n_be_coda

In summary: that's a lot of overhead for a feature that could stand in 10 python lines I think ^^ So I'm note gonna merge this PR as it, but will directly improve the code in master branch accordingly because I agree this is needed. 

Feel free to simplify your PR and re-propose it for 8, or discuss my decision.

Have a nice day ahead,
Quentin

**Comment by mart-e** [comment]
And for the red CLA, it's because you are signing your commits with a `From: =?UTF-8?q?Fran=C3=A7ois=20Honor=C3=A9?=
 <fho@Eezee-Its-MacBook-Pro.local>`

[Check your git configuration](https://help.github.com/articles/setting-your-email-in-git/) to put the email you are signing the CLA with.

**Comment by mart-e** [comment]
By the way I have merged eezee-it CLA at #11010 so just rebase your branch and commit with the correct email should make the CLA test green

**Comment by eezeefho** [comment]
Bonjour,

Je vous propose de discuter des fonctionnalités avec notre Business Analyste Janik, qui a analysé la fonctionnalité.
Car pour nous, elles sont toutes importante pour que le module soit générique et souple.

Pour que ça soit plus simple pour tout le monde, je propose une vidéo conférence.
Pouvez-vous nous transmettre des dates & plages horaires de quand vous êtes disponible pour organiser cela?


D’avance je vous remercie
[Eezee-it]<https://www.eezee-it.com/?utm_source=signature&utm_medium=mail&utm_campaign=fho><https://www.eezee-it.com/?utm_source=signature&utm_medium=mail&utm_campaign=fho>


  François Honoré  -   Odoo Developer


+32 10 87 00 24
Rue Léon Deladrière 15
1300 Limal - Belgium



www.eezee-it.com<https://www.eezee-it.com/?utm_source=signature&utm_medium=mail&utm_campaign=fho>
Follow Eezee-it!
Facebook |<https://www.facebook.com/eezee.flexware.it> Twitter |<https://twitter.com/eezeeit> LinkedIn<https://www.linkedin.com/company/eezee-it>




On 19 Feb 2016, at 11:02, qdp-odoo <notifications@github.com<mailto:notifications@github.com>> wrote:


Hi Francois, and thanks for the PR.

I've read through the diff and here is a list of things I'd like to comment:

  *   no need of compatibility check at installation: that's a bit overkill, don't bother
  *   no need of boolean 'Force Sequence' on journal: by installing the module, you already said you wanted that behavior. Keep it simple
  *   new field on bank.statement.line is a related (and should be written accordingly), but actually I don't see why you added it. It's useless, according to me, and could be simply removed
  *   I don't see any use of the demo data somewhere. I suppose you could remove them too and rely on the demo file that is already in l10n_be_coda

In summary: that's a lot of overhead for a feature that could stand in 10 python lines I think ^^ So I'm note gonna merge this PR as it, but will directly improve the code in master branch accordingly because I agree this is needed.

Feel free to simplify your PR and re-propose it for 8, or discuss my decision.

Have a nice day ahead,
Quentin

**Comment by qdp-odoo** [comment]
je passe

**Comment by eezeefho** [comment]
Excusez-moi mais je n’ai pas compris le sens de votre réponse, pouvez-vous m’éclairer?

Merci
[Eezee-it]<https://www.eezee-it.com/?utm_source=signature&utm_medium=mail&utm_campaign=fho><https://www.eezee-it.com/?utm_source=signature&utm_medium=mail&utm_campaign=fho>


  François Honoré  -   Odoo Developer


+32 10 87 00 24
Rue Léon Deladrière 15
1300 Limal - Belgium



www.eezee-it.com<https://www.eezee-it.com/?utm_source=signature&utm_medium=mail&utm_campaign=fho>
Follow Eezee-it!
Facebook |<https://www.facebook.com/eezee.flexware.it> Twitter |<https://twitter.com/eezeeit> LinkedIn<https://www.linkedin.com/company/eezee-it>








On 19 Feb 2016, at 13:54, qdp-odoo <notifications@github.com<mailto:notifications@github.com>> wrote:


je passe

**Comment by mart-e** [comment]
@eezeefho let's keep the public discussions in English please.
Sorry but we can't organise video conference for every pull request made on GitHub. If you wish to discuss the addition of a new module that does more than a simple localisation data (requires new fields, etc.), please contact support or your account manager to see what can be done.

**Comment by eezeefho** [comment]
Quentin, François,

  *   no need of boolean 'Force Sequence' on journal: by installing the module, you already said you wanted that behavior. Keep it simple
I agree, no need for the Boolean.


  *   new field on bank.statement.line is a related (and should be written accordingly), but actually I don't see why you added it. It's useless, according to me, and could be simply removed
Accountants like to keep the number the bank is using as well, especially when going from the behavior prior to this fix to the expected one. We sort the views on the external number because that’s the order where they are supposed to be recorded in (to avoid start-end balance warnings/mismatches). In fact we should put it in the header part of the statement view instead of in the tab “CODA Notes”.

While we are at it. Wouldn’t we fix the fact that this warning “The CODA Statement CLASSIC-REKENING #014 Starting Balance (xxxxx.xx) does not correspond with the previous Closing Balance (yyyyy.yy) in journal BEL27zicht!” is actually not being popped up to the user when importing a CODA file? Now it’s put in the CODA notes as well but this doesn’t give much opportunity to the user to cancel the import when it’s not in the expected order.

Thanks and kind regards,
Janik

[Eezee-it]<https://www.eezee-it.com/?utm_source=signature&utm_medium=mail&utm_campaign=jdg>

  Janik De Goÿ  -   Business Analyst

+32 10 87 00 24
Rue Léon Deladrière 15
1300 Limal - Belgium

www.eezee-it.com<https://www.eezee-it.com/?utm_source=signature&utm_medium=mail&utm_campaign=jdg>
Follow Eezee-it!
Facebook |<https://www.facebook.com/eezee.flexware.it> Twitter |<https://twitter.com/eezeeit> LinkedIn<https://www.linkedin.com/company/eezee-it>
